lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -2,6 +2,8 @@
 
 class SignupsController extends EntityAPIController {
 
+  protected $ignore_global_user = FALSE;
```

Comment:
instead of doing this, you can pass the uid into `dosomething_signup_create` or that `signup::save` and only load the `global $user` if it isn't already passed in!